flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -241,10 +264,31 @@ private void addValue(OUT reuse) throws IOException, SQLException {
 						reuse.setField(resultSet.getDouble(pos + 1), pos);
 						break;
 					case java.sql.Types.DECIMAL:
-						reuse.setField(resultSet.getBigDecimal(pos + 1).doubleValue(), pos);
-						break;
+						//TODO manage null fields
+						try {
+							if (o == null){
+								reuse.setField("", pos);
+							} else{
+								reuse.setField(resultSet.getBigDecimal(pos + 1).toPlainString(), pos);
+							}
+						} catch (SQLException e) {
+							System.err.println("error reading at position " + pos + " setting blank field!");
```

Comment:
this behavior is inconsistent with other types. either always fail or always set a default value.

In any case, such an error should be logged.